operator/src/main/java/oracle/kubernetes/operator/JobWatcher.java

```diff
@@ -283,4 +302,17 @@ public NextAction onSuccess(
           });
     }
   }
+
+  static class DeadlineExceededException extends Exception {
+    final String job;
+
+    public DeadlineExceededException(String job) {
+      super();
+      this.job = job;
+    }
+
+    public String toString() {
+      return "Job " + job + " failed. Reason: DeadlineExceeded";
```

Comment:
Could we include more information to help the user figure out the next steps?  For example:

- the deadline seconds and a hint about how to increase it  

- the number of seconds that passed since the job was submitted

- maybe something like `Ensure all domain dependencies have been deployed (any secrets, config-maps, PVs, and PVCs that the domain resource references).`  (I assume a missing dependency would be a common cause of exceeding the dead-line?)

- And finally even `kubectl describe the job and its pod for more job failure information`.

Reply:
This is good feedback; yet, I'd prefer to do this work as part of a different PR specifically around diagnosing introspector job failures.